src/blas/impl/KokkosBlas3_trsm_impl.hpp

```diff
@@ -45,16 +45,284 @@
 #define KOKKOSBLAS3_TRSM_IMPL_HPP_
 
 /// \file KokkosBlas3_trsm_impl.hpp
-/// \brief Implementation(s) of triangular linear system solve (with multiple RHSs) .
+/// \brief Implementation(s) of triangular linear system solve (with multiple RHSs)
+/// \brief Sequential fall-back implementation calls the exisiting serial batched TRSM.
+/// \brief Two sequential fall-back implementations for conjugate transpose case are
+/// \brief also based on the exisiting serial batched TRSM.
 
 #include "KokkosKernels_config.h"
 #include "Kokkos_Core.hpp"
 #include "Kokkos_ArithTraits.hpp"
+#include "KokkosBatched_Trsm_Decl.hpp"
+#include "KokkosBatched_Trsm_Serial_Impl.hpp"
 
+using namespace KokkosBatched;
 
 namespace KokkosBlas {
 namespace Impl {
-  //Note: Add an implementation of KokkosBlas::trsm in a next PR.
+
+template<typename ScalarType,
+         typename ValueType>
+int
+SerialTrsmInternalLeftLowerConj(const bool use_unit_diag,
+                                const int m, const int n,
+                                const ScalarType alpha,
+                                const ValueType *__restrict__ A, const int as0, const int as1,
+                                /**/  ValueType *__restrict__ B, const int bs0, const int bs1) {
+
+  typedef Kokkos::Details::ArithTraits<ValueType> AT;
+  
+  const ScalarType one(1.0), zero(0.0);
+      
+  if (alpha == zero)   SerialSetInternal  ::invoke(m, n, zero,  B, bs0, bs1);
+  else {
+    if (alpha != one)  SerialScaleInternal::invoke(m, n, alpha, B, bs0, bs1);
+    if (m <= 0 || n <= 0) return 0;
+
+    for (int p=0;p<m;++p) {
+      const int iend = m-p-1, jend = n;
+        
+      const ValueType
+        *__restrict__ a21 = iend ? A+(p+1)*as0+p*as1 : NULL;
```

Comment:
`__restrict__` is not a standard C++ keyword and may not build on some compilers.  I think Kokkos might have a macro you can use for this.  I can change this review if the other kokkos-kernels developers say it's OK.

Reply:
@mhoemmen Thanks. I changed them to ```KOKKOS_RESTRICT```s.

Reply:
@vqd8a thanks! :-)